File: Logic-2/makeChocolate/MakeChocolate.java
package makeChocolate;

public class MakeChocolate {
    //Retorne o número de "small" que seja necessário para atingir o "goal", sabendo que o "big" será usando antes do "small"
    //e que cada "small" vale 1 e cada "big" vale 5. Caso não dê para atingir o "goal" retorne -1.
    //Ex.: ((4, 1, 7) → 2; (4, 1, 10) → -1).

    public int makeChocolate(int small, int big, int goal) {
        int totBig = 5*big;
        if (totBig > goal) totBig = (goal/5)*5;
        if (small + totBig >= goal) return goal-totBig;
        else return -1;
    }
}

File: Python/Logic-2/make_chocolate.py
"""
Sabendo que cada unidade de "small" vale 1 e que cada unidade de "big" vale 5,
retorne a quantidade de "small" necessária para se atingir o "goal", sabendo que
deverá se usar todas as "big" possíveis antes. Caso não dê para atingir o "goal",
retorne -1.
Ex.:((4, 1, 9) → 4; (4, 1, 10) → -1; (4, 1, 7) → 2).
"""


def make_chocolate(small, big, goal):
    goal -= int(goal/5)*5 if big*5 > goal else big * 5
    if goal > small:
        return -1
    else:
        return goal


print(make_chocolate(4, 1, 9))
